src/visualization/utils/useStaticLegend.ts

```diff
@@ -79,6 +84,23 @@ export const useStaticLegend = (properties): StaticLegendConfig => {
     },
     [dispatch, properties.staticLegend]
   )
+
+  let updateStaticLegendProperties
+  const notebookViewProperties = data?.properties || {}
+  const notebookStaticLegend = notebookViewProperties.staticLegend || {}
+  if (id) {
+    updateStaticLegendProperties = (updatedProperties: StaticLegendAPI) =>
+      update({
+        ...data,
```

Comment:
you might have issues with stale data by implementing the spread operator here if multiple update requests are made in the same render cycle, so i'd delete this line. the same'll happen for the properties as well, but we currently dont have a mechanism for mitigating nested update collisions as it'll happen at less of a chance.

Reply:
Thanks, I'll remove `...data`.

Yeah, the `properties` update, we will have to take a chance on. Human users are probably not fast enough to cause the collision from selecting different view properties.